source/client/service_impl.cc

```diff
@@ -0,0 +1,89 @@
+#include "client/service_impl.h"
+
+#include <grpc++/grpc++.h>
+
+#include "client/client.h"
+#include "client/options_impl.h"
+
+namespace Nighthawk {
+namespace Client {
+
+void ServiceImpl::handleExecutionRequest(const nighthawk::client::ExecutionRequest& request) {
+  OptionsPtr options;
+  try {
+    options = std::make_unique<OptionsImpl>(request.start_request().options());
+  } catch (Envoy::EnvoyException exception) {
+    response_queue_.push_back(ServiceProcessResult({}, exception.what()));
+    return;
+  }
+
+  auto logging_context = std::make_unique<Envoy::Logger::Context>(
+      spdlog::level::from_str(options->verbosity()), "[%T.%f][%t][%L] %v", log_lock_);
+  process_ = std::make_unique<ProcessImpl>(*options, time_system_);
+  OutputCollectorFactoryImpl output_format_factory(time_system_, *options);
+  auto formatter = output_format_factory.create();
+  bool success = process_->run(*formatter);
+  nighthawk::client::ExecutionResponse response;
+  *(response.mutable_output()) = formatter->toProto();
+  response_queue_.emplace_back(ServiceProcessResult(response, success ? "" : "Unkown failure"));
+  process_.reset();
```

Comment:
Can `process_` be scoped to just this method; we won't need an explicit reset in that case.

Reply:
Done